selfdrive/ui/sunnypilot/qt/offroad/settings/vehicle_panel.cc

```diff
@@ -20,13 +24,44 @@ VehiclePanel::VehiclePanel(QWidget *parent) : QFrame(parent) {
   platformSelector = new PlatformSelector();
   list->addItem(platformSelector);
 
+  QObject::connect(uiState(), &UIState::offroadTransition, this, [=](bool offroad_transition) {
+    updateToggles(offroad_transition);
+    updatePanel(offroad_transition);
+  });
+
   ScrollViewSP *scroller = new ScrollViewSP(list, this);
   vlayout->addWidget(scroller);
 
-  QObject::connect(uiState(), &UIState::offroadTransition, this, &VehiclePanel::updatePanel);
+  QObject::connect(platformSelector, &PlatformSelector::clicked, [=]() {
+    QTimer::singleShot(100, this, &VehiclePanel::updateCarToggles);
+  });
+
+  std::vector<QString> tuning_buttons { tr("Off"), tr("Long Tune"), tr("Tune + Even Smoother Braking") };
+  hkgtuningToggle = new ButtonParamControlSP(
+    "HyundaiLongTune",
+    tr("HKG Custom Longitudinal Tuning"),
+    tr("Select a tuning mode. 'Off' means there is no custom tuning is currently applied. "
+       "'Long Tune' is a dynamic acceleration/brake tune individualized to your specific car. "
+       "'Tune + Smoother Braking' is the dynamic tuning but with even smoother braking applied."),
+       "../assets/offroad/icon_shell.png",
+    tuning_buttons
+  );
+  hkgtuningToggle->showDescription();
+  hkgtuningToggle->setProperty("originalDesc", hkgtuningToggle->getDescription());
+
+  connect(hkgtuningToggle, &ButtonParamControlSP::buttonToggled, this, [=](int index) {
+    hkg_state = index;
+    // Inline update of tuning and braking params
+    params.put("HyundaiLongTune", QString::number(index).toStdString());
+    params.put("HyundaiSmootherBraking", (index == 2) ? "1" : "0");
```

Comment:
`index == 2` doesn't feel nice to read, I'd go for an enum here so that it's readable, and I'd use `params.putBool`

so something like:

```
params.put("HyundaiSmootherBraking", index == IncludeSmootherBraking);
```